chapters/common_pitfalls.md

```diff
@@ -0,0 +1,44 @@
+
+### Vulkan is a box of tools
+
+In Vulkan, there are often multiple solutions to a problem. Consider passing data to a shader, one can use uniform buffers, storage buffers, vertex attributes, instance buffer, push constants, and more. Are all valid methods each with their own benefits and drawbacks. There is rarely a "perfect" solution to any problem. Therefore, use the tools available and create a good solution instead of trying to find a perfect solution.
+
+### Recording command buffers  
+
+A common assumption when first learning about command buffers is that reusing them is paramount and that re-recording them every frame is costly. While it may appear counterintuitive, there is often a greater cost in reuse than simply recording new command buffers each frame. The main cost is the additional complexity for managing dynamic objects, frustum culling, and many other graphical effects. Many of the approaches used to reduce command buffer recording add non trivial state management or undue complexity to the rendering architecture. Therefore, especially in simpler situations, the cost of re-recording every frame is reasonable. Developers still concerned can and should profile command buffer recording to make better informed decisions about the best approach for their application.
+
+### Multiple pipelines
+
+A graphics `VkPipeline` contains the combination of state needed to perform a draw call. For every input combination (shaders, vertex layout, primitive assembly, depth testing, blending mode, etc), a new pipeline is needed. This necessitates creating and binding many pipelines for complex rendering situations. While it might appear beneficial to try to create as few pipeline as possible by increasing complexity elsewhere, it is best to profile and determine if this is truly a big performance penalty in the first place.
+
+### Resource duplication per swapchain image
+
+A common technique to increase throughput is to make multiple instances of a resource, where each instance can be used while rendering a different frame. The swapchain is one such example, as it contains multiple images. This can lead to the assumption that the number of instances each resource has should be the same as the swapchain image count. The issue with this assumption is that very few resources need to be duplicated the same amount as the swapchain image count. While it is useful for some things, for example the number of semaphores used to signal when a swapchain image is available, it is worth understanding which resources have one to one relationships and which don't.
```

Comment:
I think this needs a sentence or two explaining that you only need one of a given resource if all uses are temporally separated - e.g. if a depth buffer is used in one frame, then the data is never used again after that frame, then the next frame can use the same depth buffer.

Reply:
The idea of "temporally separated" is trickier with pipelined GPUs, fragment work may run in parallel with vertex work for the following frame. The depth buffer example still holds true because it only involves fragment and depth tests, but if reusing a resource creates a FRAG -> VERT dependency, that would create bubbles in the pipeline.

Reply:
This is the one point that is rather hard to write because there is a lot of nuance and jargon at play. 
My main point succinctly put is "Few resources need to be duplicated the same amount as the swapchain image count". IE, I have X swapchain images, therefore I should have X everything else. 
There are so many ways to solve the topic of resource contention and pipelining frames that it deserves its own page. As thus I am just trying to highlight that not everything requires 1:1.